src/components/CheckBoxList/check-box-list.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { useState } from 'react';
+import { Box } from '@mui/material';
+import { DragDropContext, Droppable } from 'react-beautiful-dnd';
+import { CheckBoxListItems } from './check-box-list-items';
+import { CheckboxListProps } from './check-box-list-type';
+
+export function CheckboxList<T>({
+    items,
+    selectedItems,
+    getItemId,
+    getItemLabel,
+    sx,
+    secondaryAction,
+    enableSecondaryActionOnHover = true,
+    isDisabled,
+    onSelectionChange,
+    addSelectAllCheckbox = false,
+    isDndDragAndDropActive = false,
+    onDragStart,
+    onDragEnd,
+    isDragDisable = false,
+    divider,
+    ...props
```

Comment:
```suggestion
    enableSecondaryActionOnHover = true,
    addSelectAllCheckbox = false,
    isDndDragAndDropActive = false,
    onDragStart,
    onDragEnd,
    isDragDisable = false,
    ...props
```
we are only passing props, no need to destructure them
you can even remove everything except onDragEnd and onDragStart, then define default values in <CheckBoxListItems> component